runtime/lua/vim/_editor.lua

```diff
@@ -446,10 +446,17 @@ do
   ---@param msg string Content of the notification to show to the user.
   ---@param level number|nil One of the values from |vim.log.levels|.
   ---@param opts table|nil Optional parameters. Unused by default.
-  function vim.notify_once(msg, level, opts) -- luacheck: no unused args
-    if not notified[msg] then
+  function vim.notify_once(msg, level, opts, key)
```

Comment:
This could return `true` if it displays and `false` if it skipped. Then `vim.deprecate` could use that to decide whether it should show a backtrace (using plain `vim.notify()` or whatever?).

That keeps the door open for making backtrace optional. Would this work?

Reply:
Sure. Should we add an optional `opts` dictionary to `vim.deprecate` as the fifth argument then?

Reply:
> Sure. Should we add an optional opts dictionary to vim.deprecate as the fifth argument then?

In general when a Lua function wants to have an `opts` arg we should almost always change the function to accept a "kwargs" form (i.e. a single dict param). The old non-dict params form remains as an optional interface.

So, for now I'd say just add another param, unless you want to take a stab at that.